packages/mds-types/transformers/0_4_1_to_1_0_0/index.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  VEHICLE_REASON_0_4_1,
+  VehicleEvent_v0_4_1,
+  VEHICLE_EVENT_0_4_1,
+  TRANSFORMER_VEHICLE_EVENT,
+  TRANSFORMER_EVENT_TYPE_REASON
+} from '../@types'
+import { VEHICLE_EVENT, VehicleEvent, VEHICLE_STATE } from '../../index'
+
+export const FULL_STATE_MAPPING_0_4_1_to_1_0_0: {
+  [P in VEHICLE_EVENT_0_4_1 | TRANSFORMER_VEHICLE_EVENT]: {
+    [Q in VEHICLE_REASON_0_4_1 | TRANSFORMER_EVENT_TYPE_REASON]: {
+      event_type: VEHICLE_EVENT
+      vehicle_state: VEHICLE_STATE
+    }
+  }
+} = {
+  /* `agency_drop_off` included for the sake of completeness. No provider will ever submit such
+   * an event type through the agency API.
+   */
+  agency_drop_off: { no_event_type_reason: { event_type: 'agency_drop_off', vehicle_state: 'available' } },
+  agency_pick_up: { no_event_type_reason: { event_type: 'agency_pick_up', vehicle_state: 'removed' } },
+  cancel_reservation: { no_event_type_reason: { event_type: 'reservation_cancel', vehicle_state: 'available' } },
+  deregister: {
+    decommissioned: { event_type: 'decommissioned', vehicle_state: 'removed' },
+    missing: { event_type: 'missing', vehicle_state: 'removed' }
+  },
```

Comment:
For the entries which have defined `event_type_reason`s, are we _certain_ they will? Feels like it'd be safest to support the no-event-type-reason case for everything.

Reply:
agree

Reply:
Well, we used to have validation that took care of that. I think bringing back the 0.4.1 validators in some form would be good.

Reply:
Validators brought back. Will discuss with @avatarneil how to use them, but my reasoning for bringing them back is to preserve as much 0.4.1 behavior as possible.

Reply:
Didn't realize we don't actually use the validators for anything. NVM.